### PR TITLE
fix: make terminal work after stopping and starting a container

### LINKED ISSUE #8363: Terminal stops working when the container is stopped and started again
### Bug description

Reported via user's feedback:
> Comment: If the container is stopped and restarted. And then i click terminal, its the old defunct terminal! Please fix!
Contact:  in US
Running 1.11.1 on Mac, arm64, 24.0.0

### Operating system

Fedora

### Installation Method

Other

### Version

next (development version)

### Steps to reproduce

1. Start a container (`ghcr.io/linuxcontainers/alpine`)
2. Go to the terminal and type a command, ie: `ls`
3. Stop the container
4. Start the container again
5. Go to terminal
6. Assert: type `whoami` (any command) in the terminal and verify that you get the response
Actual state: Terminal is not responding, you can see previous command (`ls`) execution.

Note: If you restart the container, it seems to work, just stop+start action looks like causing the issue

### Relevant log output

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by SoniaSandler**
Signed-off-by: Sonia Sandler <ssandler@redhat.com>

### What does this PR do?
Fixes container terminal to be responsive after stopping and starting a container

### Screenshot / video of UI
[Screencast from 2024-08-29 11-09-56.webm](https://github.com/user-attachments/assets/78fa006e-50d2-46cc-b082-7da29b7163a1)

### What issues does this PR fix or reference?
Closes https://github.com/containers/podman-desktop/issues/8363

### How to test this PR?

- [ ] Tests are covering the bug fix or the new feature

### CODE REVIEW

**Review by feloy — COMMENTED** [review]
Works, thanks.
A unit test would be helpful

File: packages/renderer/src/lib/container/ContainerDetailsTerminal.spec.ts [hunk 1]
```diff
@@ -112,3 +114,78 @@ test('expect being able to reconnect ', async () => {
...
+
+  await renderObject.rerender({ container: container, screenReaderMode: true });
+
+  await tick();
+
+  await new Promise(resolve => setTimeout(resolve, 100));
+
+  container.state = 'RUNNING';
+
+  console.log(container);
```
Comment (benoitf) [inline packages/renderer/src/lib/container/ContainerDetailsTerminal.spec.ts]: to be removed
```suggestion
```

File: packages/renderer/src/lib/container/ContainerDetailsTerminal.spec.ts [hunk 2]
```diff
@@ -112,3 +114,78 @@ test('expect being able to reconnect ', async () => {
...
+  await renderObject.rerender({ container: container, screenReaderMode: true });
+
+  await tick();
+
+  await new Promise(resolve => setTimeout(resolve, 100));
+
+  expect(screen.queryByText('Container is not running')).toBeInTheDocument();
+
+  container.state = 'STARTING';
+  console.log(container);
```
Comment (benoitf) [inline packages/renderer/src/lib/container/ContainerDetailsTerminal.spec.ts]:
```suggestion
```

File: packages/renderer/src/lib/container/ContainerDetailsTerminal.spec.ts [hunk 3]
```diff
@@ -112,3 +114,78 @@ test('expect being able to reconnect ', async () => {
...
+  await new Promise(resolve => setTimeout(resolve, 1000));
+
+  // search a div having aria-live="assertive" attribute
+  const terminalLinesLiveRegion = renderObject.container.querySelector('div[aria-live="assertive"]');
+
+  // check the content
+  expect(terminalLinesLiveRegion).toHaveTextContent('hello world');
+
+  container.state = 'EXITED';
+  console.log(container);
```
Comment (benoitf) [inline packages/renderer/src/lib/container/ContainerDetailsTerminal.spec.ts]:
```suggestion
```

File: packages/renderer/src/lib/container/ContainerDetailsTerminal.spec.ts [hunk 4]
```diff
@@ -112,3 +114,78 @@ test('expect being able to reconnect ', async () => {
...
+  await waitFor(() => expect(shellInContainerMock).toHaveBeenCalled());
+
+  // write some data on the terminal
+  onDataCallback(Buffer.from('hello\nworld'));
+
+  // wait 1s
+  await new Promise(resolve => setTimeout(resolve, 1000));
+
+  // search a div having aria-live="assertive" attribute
+  const terminalLinesLiveRegion = renderObject.container.querySelector('div[aria-live="assertive"]');
```
Comment (benoitf) [inline packages/renderer/src/lib/container/ContainerDetailsTerminal.spec.ts]: need to be in a waitFor block

File: packages/renderer/src/lib/container/ContainerDetailsTerminal.spec.ts [hunk 5]
```diff
@@ -112,3 +114,78 @@ test('expect being able to reconnect ', async () => {
...
+  container.state = 'EXITED';
+  console.log(container);
+
+  await renderObject.rerender({ container: container, screenReaderMode: true });
+
+  await tick();
+
+  await new Promise(resolve => setTimeout(resolve, 100));
+
+  expect(screen.queryByText('Container is not running')).toBeInTheDocument();
```
Comment (benoitf) [inline packages/renderer/src/lib/container/ContainerDetailsTerminal.spec.ts]: need to be in waitFor

also it's strange if tick is not doing the update ?
missing a new tick() ?

File: packages/renderer/src/lib/container/ContainerDetailsTerminal.spec.ts [hunk 6]
```diff
@@ -112,3 +114,78 @@ test('expect being able to reconnect ', async () => {
...
+
+  container.state = 'STARTING';
+  console.log(container);
+
+  await renderObject.rerender({ container: container, screenReaderMode: true });
+
+  await tick();
+
+  await new Promise(resolve => setTimeout(resolve, 100));
```
Comment (benoitf) [inline packages/renderer/src/lib/container/ContainerDetailsTerminal.spec.ts]: to be removed

File: packages/renderer/src/lib/container/ContainerDetailsTerminal.spec.ts [hunk 7]
```diff
@@ -112,3 +114,78 @@ test('expect being able to reconnect ', async () => {
...
+
+  container.state = 'RUNNING';
+
+  console.log(container);
+
+  await renderObject.rerender({ container: container, screenReaderMode: true });
+
+  await tick();
+
+  await new Promise(resolve => setTimeout(resolve, 100));
```
Comment (benoitf) [inline packages/renderer/src/lib/container/ContainerDetailsTerminal.spec.ts]: strange if we wait and again waitFor ?

**Review by feloy — APPROVED** [review]

**Review by benoitf — APPROVED** [review]

**Pull request merged**